Add Check Support for Java 21 Pattern Matching for Switch Syntax: NestedIfDepth

child of https://github.com/checkstyle/checkstyle/issues/14961

I have read check documentation:https://checkstyle.org/checks/coding/nestedifdepth.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> cat src/Test.java  
record ColoredPoint(int p, int x, String c) { }
record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { }

public class Test {

    void test(Object obj) {
        switch (obj) {
            case ColoredPoint(int x, int y, String z) when (x >= 2)
                    -> {
                if (x >= 2) {
                    if(y >= 2) {
                        if (z.equals("red")) {

                        }
                    }
                }
            }
            default -> {}
        }
    }
}
PS D:\CS\test> cat config.xml                                                   
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="NestedIfDepth">

        </module>
    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:12:25: Nested if-else depth is 2 (max allowed is 1). [NestedIfDepth]
Audit done.
Checkstyle ends with 1 errors.

```

---
Describe what you want in detail

`when` can't be nested and nested `ifs` in the switch blocks work correctly. We are ok here. This can be closed.

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/NestedIfDepthCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/NestedIfDepthCheckTest.java
@@ -59,6 +59,17 @@ public class NestedIfDepthCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testInsideCaseBody() throws Exception {
+
+        final String[] expected = {
+            "17:25: " + getCheckMessage(MSG_KEY, 2, 1),
+        };
+
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputNestedIfDepthInsideCaseBody.java"), expected);
+    }
+
+    @Test
     public void testTokensNotNull() {
         final NestedIfDepthCheck check = new NestedIfDepthCheck();
         assertWithMessage("Acceptable tokens should not be null")

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/nestedifdepth/InputNestedIfDepthInsideCaseBody.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/nestedifdepth/InputNestedIfDepthInsideCaseBody.java
@@ -1,0 +1,29 @@
+/*
+NestedIfDepth
+depth = (default)1
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.nestedifdepth;
+
+public class InputNestedIfDepthInsideCaseBody {
+
+    public void test(Object obj) {
+        switch (obj) {
+            case ColoredPoint(int x, int y, String z) when (x >= 2) -> {
+                if (x >= 2) {
+                    if (y >= 2) {
+                        if (z.equals("red")) { // violation
+
+                        }
+                    }
+                }
+            }
+            default -> {
+            }
+        }
+    }
+
+    record ColoredPoint(int x, int y, String z) { }
+}

